$NetBSD: patch-ab,v 1.6 2010/09/23 07:07:47 adam Exp $

Log all pkg-config calls to a file.  Used by pkgtools/verifypc.

--- main.c.orig	2010-05-10 17:58:57.000000000 +0000
+++ main.c
@@ -462,12 +462,27 @@ main (int argc, char **argv)
     gboolean failed = FALSE;
     GSList *reqs;
     GSList *iter;
+    char *log_name;
+    FILE *log;
 
     reqs = parse_module_list (NULL, str->str,
                               "(command line arguments)");
 
     iter = reqs;
 
+    log_name = getenv ("PKG_CONFIG_LOG");
+    if (log_name != NULL)
+      {
+        log = fopen (getenv ("PKG_CONFIG_LOG"), "a");
+        if (log == NULL)
+          {
+            fprintf (stderr, "Cannot open log file: %s\n", log_name);
+            exit (1);
+          }
+      }
+    else
+      log = NULL;
+
     while (iter != NULL)
       {
         Package *req;
@@ -489,6 +504,18 @@ main (int argc, char **argv)
 	    fprintf (log, "\n");
 	  }
 
+        if (log != NULL)
+          {
+            if (req == NULL)
+              fprintf (log, "%s NOT-FOUND", ver->name);
+            else
+              fprintf (log, "%s %s %s", ver->name,
+                       comparison_to_str (ver->comparison),
+                       (ver->version == NULL) ? "(null)" : ver->version);
+
+            fprintf (log, "\n");
+          }
+
         if (req == NULL)
           {
             failed = TRUE;
@@ -524,6 +551,11 @@ main (int argc, char **argv)
 	fclose (log);
       }
 
+    if (log != NULL)
+      {
+        fclose (log);
+      }
+
     if (failed) {
       return 1;
     }
